datafusion/core/src/config.rs

```diff
@@ -0,0 +1,187 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! DataFusion Configuration Options
+
+use datafusion_common::ScalarValue;
+use sqlparser::ast::DataType;
+use std::collections::HashMap;
+
+/// Configuration option "datafusion.optimizer.filterNullJoinKeys"
+pub const OPT_FILTER_NULL_JOIN_KEYS: &str = "datafusion.optimizer.filterNullJoinKeys";
+
+/// Definition of a configuration option
+pub struct ConfigDefinition {
+    /// key used to identifier this configuration option
+    key: String,
+    /// Description to be used in generated documentation
+    description: String,
+    /// Data type of this option
+    data_type: DataType,
+    /// Default value
+    default_value: ScalarValue,
+}
+
+impl ConfigDefinition {
+    /// Create a configuration option definition
+    pub fn new(
+        name: &str,
+        description: &str,
```

Comment:
```suggestion
        name: impl Into<String>,
        description: impl Into<String>,
```

And then you can change `name.to_string()` to `name.into()` and `description.to_string()` into `description.into()`

It is a minor thing, but that way the caller can pass in a `String` if they have it and avoid the extra copy 🤷